src/KeyManager.cpp

```diff
@@ -4,59 +4,58 @@
 #include <iostream>
 
 KeyManager::KeyManager() {
-    refreshKeys();
+	refreshKeys();
 }
 
 void KeyManager::refreshKeys() {
-    std::vector<AccessKeys> keys;
-    std::string contents;
-    for (const auto& entry : std::filesystem::directory_iterator("../keys")) {
-        std::string filename = entry.path().string();
-        readFile(contents, filename);
-        if (contents.length() != HASH_LENGTH * 2 + 1) {
-            std::cerr << "Reading hash file with incorrect contents... " << filename << std::endl;
-            continue;
-        }
-
-        const char* contentCharacters = contents.data();
-        std::array<char, HASH_LENGTH> moonKey {};
-        std::array<char, HASH_LENGTH> playerKey {};
-        for (int i = 0; i < HASH_LENGTH; i++) {
-            moonKey[i] = contentCharacters[i];
-            playerKey[i] = contentCharacters[i + HASH_LENGTH + 1];
-        }
-
-        keys.emplace_back(AccessKeys {
-                moonKey,
-                playerKey
-        });
-        contents.clear();
-    }
-
-    const std::lock_guard<std::mutex> keyLock(m_keyMutex);
-    m_keys.swap(keys);
+	std::vector<AccessKeys> keys;
+	std::string contents;
+	std::string filename;
+	for (const auto& entry : std::filesystem::directory_iterator("../keys")) {
+		filename = entry.path().string();
+		readFile(contents, filename);
+		if (contents.length() != HASH_LENGTH * 2 + 1) {
+			std::cerr << "Reading hash file with incorrect contents... " << filename << std::endl;
+			continue;
+		}
```

Comment:
however this stuff should be reverted because this is actually causing additional copies and allocations. everytime you do this `filename = entry.path().string();` its exactly the same as doing this `std::string filename = entry.path().string();` except worse because you also initialize string above...

in reality these are going to be exactly the same performance wise but technically the old version is better